src/pages/CohortManagerMainPage.jsx

```diff
@@ -0,0 +1,139 @@
+import "../App.css";
+import Post from "../components/Post";
+import { PostContext } from "../App.jsx";
+import { useContext, useEffect, useState } from "react";
+import { getAContactByID, makeNewPostToAPI } from "../Api.js";
+import {
+  ProfileIcon,
+  HomePageIcon,
+  MainPageIcon,
+} from "../components/images.jsx";
+import GetInitalsFromNames from "../GetInitialsFromNames.jsx";
+
+const InitialPostData = "";
+
+function CohortManagerMainPage() {
+  const [simulatedUserData, setSimulatedUserData] =
+    useState(null); /* simulating a user with id 1 for posting  */
+
+  const { postsData, setPostsData } = useContext(PostContext);
+  const [newPostData, setNewPostData] = useState(InitialPostData);
+
+  //Controls whether the post button should be disabled or not.
+  useEffect(() => {
+    getAContactByID(
+      1,
+      setSimulatedUserData
+    ); /* simulating a user with id 1 for posting  */
+
+    //Moved this check out of stringToPostRequestBody. Instead of checking for a bad value and sending an error, I decided that the button should not be interactable while the postData is bad/empty.
+    console.log("doing this");
+    console.log(newPostData);
+    if (
+      newPostData === "" ||
+      newPostData === null ||
+      newPostData === undefined ||
+      !newPostData.includes("|")
+    ) {
+      console.log("you are not allowed to make an empty post");
+      document.getElementById("submit-button").disabled = "disabled";
```

Comment:
Direct DOM manipulation in React applications (e.g., document.getElementById("submit-button").disabled = "disabled") is not recommended because it bypasses React's declarative update mechanism. Instead, you can manage the disabled state of the button reactively through state